### PR TITLE
helm: expose optional workspace paths while deploying the cluster

### PR DESCRIPTION
**Pull request by marcdiazsan**
Closes reanahub/reana#532

### CODE REVIEW

File: helm/reana/templates/reana-workflow-controller.yaml [hunk 1]
```diff
@@ -248,3 +263,10 @@ spec:
           path: {{ $workspace_path._0 }}
       {{- end }}
       {{- end }}
+      {{- if .Values.workspaces }}{{ if .Values.workspaces.paths }}
+      {{- range $workspace_paths := .Values.workspaces.paths }}
+      - name: {{ $workspace_paths | quote | replace "/" "" }}
+        hostPath:
+          path: {{ $workspace_paths }}
+      {{- end }}
+      {{- end }}{{ end }}
```
Comment (audrium) [inline helm/reana/templates/reana-workflow-controller.yaml:272]: Minor: empty line is missing

File: reana/reana_dev/cluster.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ def cluster_build(
     "--workspace-mounts",
     multiple=True,
     help="Which directories from the Kubernetes nodes to mount inside the cluster pods? "
-    "cluster_node_path:job_pod_mountpath, e.g /var/reana/mydata:/mydata",
+    "cluster_node_path:cluster_pod_mountpath, e.g /var/reana/mydata:/mydata",
```
Comment (audrium) [inline reana/reana_dev/cluster.py:239]: Can you explain why this was needed? `/mydata` is `job_pod_mountpath`

File: helm/reana/templates/reana-server.yaml [hunk 3]
```diff
@@ -78,6 +85,10 @@ spec:
             value: {{ .Values.quota.default_disk_limit | default 0 | quote }}
           - name: REANA_KUBERNETES_JOBS_MEMORY_LIMIT
             value: {{ .Values.kubernetes_jobs_memory_limit | default "4Gi" }}
+          {{- if .Values.workspaces }}{{ if .Values.workspaces.paths }}
+          - name: ADMIN_ALLOWED_WORKSPACES
```
Comment (audrium) [inline helm/reana/templates/reana-server.yaml:89]: Maybe `WORKSPACE_PATHS` would be more clear? WDYT?

**Review by audrium — COMMENTED** [review]
Works nicely, just left couple of minor comments

File: reana/reana_dev/cluster.py (hunk 2 above)
Comment (marcdiazsan) [reply reana/reana_dev/cluster.py:239]: But this paths are mounted not only for the jobs, but also for cluster pods `r-w-controller` and `reana-server`. I thought that `job_pod_mountpath` was not completely correct (but cluster_pod_mountpath might also not be correct, as is not for all cluster pods.....).

File: helm/reana/templates/reana-workflow-controller.yaml [hunk 4]
```diff
@@ -248,3 +263,11 @@ spec:
...
+
```
Comment (audrium) [inline helm/reana/templates/reana-workflow-controller.yaml:273]: Extremely minor: there are some spaces in the last line, that's why it still shows `No new line` error icon

File: helm/reana/values.yaml [hunk 5]
```diff
@@ -11,6 +11,11 @@ debug:
 eos:
   enabled: false
 
+workspaces:
+  paths:
+    - /tmp/reana
+    - /tmp/foo
```
Comment (audrium) [inline helm/reana/values.yaml:17]: This is nice for testing, but perhaps should be removed in the final PR version?

Comment (marcdiazsan) [reply helm/reana/values.yaml:17]: Oops, leftovers :sweat_smile:

**Review by audrium — APPROVED** [review]
Works nicely!

**Pull request merged**